modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/rest/SW360AuthenticationClient.java

```diff
@@ -43,53 +44,41 @@ public SW360AuthenticationClient(String authServerUrl) {
         this.authServerUrl = authServerUrl;
     }
 
-    public String getOAuth2AccessToken(String username, String password) throws AntennaException {
+    public String getOAuth2AccessToken(String username, String password, String clientId, String clientPassword)
+            throws AntennaException {
         String requestUrl = authServerUrl + GET_ACCESS_TOKEN_ENDPOINT;
 
-        UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(requestUrl)
-                .queryParam(SW360Attributes.AUTHENTICATOR_GRANT_TYPE, GRANT_TYPE_VALUE)
-                .queryParam(SW360Attributes.AUTHENTICATOR_USERNAME, username)
-                .queryParam(SW360Attributes.AUTHENTICATOR_PASSWORD, password);
+        String body = String.format("%s=%s&%s=%s&%s=%s", SW360Attributes.AUTHENTICATOR_GRANT_TYPE, GRANT_TYPE_VALUE,
+                SW360Attributes.AUTHENTICATOR_USERNAME, username, SW360Attributes.AUTHENTICATOR_PASSWORD, password);
 
         HttpHeaders headers = new HttpHeaders();
-            addBasicAuthentication(headers, LIFERAY_CLIENT_CREDENTIALS);
-
-            HttpEntity<String> httpEntity = new HttpEntity<>(headers);
-            ResponseEntity<String> response = this.restTemplate
-                    .exchange(builder.toUriString(),
-                            HttpMethod.POST,
-                            httpEntity,
-                            String.class);
-            if (response.getStatusCode() == HttpStatus.OK) {
-                try {
-                    return (String) new ObjectMapper().readValue(response.getBody(), HashMap.class).get("access_token");
-                } catch (IOException e) {
-                    throw new AntennaException("Error when attempting to deserialise the response body.", e);
-                }
-            } else {
-                throw new AntennaException("Could not request OAuth2 access token for [" + username + "].");
+        addBasicAuthentication(headers, clientId, clientPassword);
+        headers.add(HttpHeaders.CONTENT_TYPE, "application/x-www-form-urlencoded");
+
+        HttpEntity<String> httpEntity = new HttpEntity<>(body, headers);
+        Logger logger = LoggerFactory.getLogger(getClass());
+        logger.debug("URL: " + requestUrl);
+        logger.debug("Entity: " + httpEntity);
```

Comment:
this logs the credentials (in debug mode)?

Reply:
Same applies here, also this has been removed.